src/hotspot/cpu/ppc/sharedRuntime_ppc.cpp

```diff
@@ -2463,9 +2464,18 @@ nmethod *SharedRuntime::generate_native_wrapper(MacroAssembler *masm,
     __ addi(r_box, R1_SP, lock_offset);
 
     // Try fastpath for locking.
-    // fast_lock kills r_temp_1, r_temp_2, r_temp_3.
-    __ compiler_fast_lock_object(CCR0, r_oop, r_box, r_temp_1, r_temp_2, r_temp_3);
-    __ beq(CCR0, locked);
+    if (LockingMode == LM_LIGHTWEIGHT) {
+      // TODO: Is the fast_lock optimization needed for native wrappers?
+      //       Deviates from other platforms.
+      Label slow;
+      __ lightweight_lock(r_oop, r_temp_1, r_temp_2, slow);
+      __ b(locked);
+      __ bind(slow);
+    } else {
+      // fast_lock kills r_temp_1, r_temp_2, r_temp_3.
+      __ compiler_fast_lock_object(CCR0, r_oop, r_box, r_temp_1, r_temp_2, r_temp_3);
+      __ beq(CCR0, locked);
+    }
```

Comment:
I appreciate your effort to make it more similar among all platforms. However, I think that using the C2 version is better because the native wrapper is the highest tier. Can we keep it as before with this PR and use a separate issue to clean it up for all platforms?
That will allow us to check performance impact individually. Mixing 2 performance related changes is not ideal.

Reply:
I changed it so that the native wrapper also uses the same fast lock/unlock as the C2 code. 
Added the assert code which checks that lock stack does not contain the object when handling the inflated case. Would be nice to run some stress tests and validate the assumptions made by the fast lock/unlock.

Reply:
Thanks! I'll take a look and run tests. The assertion makes sense anyway.